File: SYP.Web/Modules/Administration/AuditLog/AuditLogColumns.cs
namespace SYP.Administration.Columns;

[ColumnsScript("Administration.AuditLog")]
[BasedOnRow(typeof(AuditLogRow), CheckNames = true)]
public class AuditLogColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight, Width(80)]
    public long Id { get; set; }

    [Width(150)]
    public DateTime ActionDate { get; set; }

    [Width(100)]
    public string ActionType { get; set; }

    [Width(150)]
    public string EntityType { get; set; }

    [EditLink, Width(200)]
    public string EntityName { get; set; }

    [Width(100)]
    public string EntityId { get; set; }

    [Width(120)]
    public string Username { get; set; }

    [Width(120)]
    public string IpAddress { get; set; }

    [Width(200)]
    public string ChangedFields { get; set; }
}

File: SYP.Web/Modules/ServerTypes/Administration/AuditLogColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { AuditLogRow } from "./AuditLogRow";

export interface AuditLogColumns {
    Id: Column<AuditLogRow>;
    ActionDate: Column<AuditLogRow>;
    ActionType: Column<AuditLogRow>;
    EntityType: Column<AuditLogRow>;
    EntityName: Column<AuditLogRow>;
    EntityId: Column<AuditLogRow>;
    Username: Column<AuditLogRow>;
    IpAddress: Column<AuditLogRow>;
    ChangedFields: Column<AuditLogRow>;
}

export class AuditLogColumns extends ColumnsBase<AuditLogRow> {
    static readonly columnsKey = 'Administration.AuditLog';
    static readonly Fields = fieldsProxy<AuditLogColumns>();
}